### ISSUE TITLE
Reducci-n-de-dimensionalidad-t-cnica-ICA--Luciano-Hernan-Garcia-Huaricallo

### ISSUE THREAD

**Issue by hlgh164469** [frdtorres/mainf24-2-BIDA#3]
import requests
import pandas as pd
def obtener_datos_api(url):
    respuesta = requests.get(url)
    if respuesta.status_code == 200:
        return respuesta.json()  
    else:
        raise Exception(f"Error al obtener datos: {respuesta.status_code}")

url_api = 'https://www.topitop.pe/hombre/casacas-hombre' 
datos = obtener_datos_api(url_api)

df = pd.DataFrame(datos)
print(df.head())  

df = df.dropna()  # Eliminar filas con valores nulos
df_normalizado = (df - df.mean()) / df.std()  # Normalización
from sklearn.decomposition import FastICA

n_componentes = 2  

# Aplicar ICA
ica = FastICA(n_components=n_componentes)
componentes_independientes = ica.fit_transform(df_normalizado)

df_ica = pd.DataFrame(componentes_independientes, columns=[f'Componente_{i+1}' for i in range(n_componentes)])
print(df_ica.head())  
import matplotlib.pyplot as plt

plt.figure(figsize=(10, 5))
plt.scatter(df_ica['Componente_1'], df_ica['Componente_2'])
plt.title('Componentes Independientes')
plt.xlabel('Componente 1')
plt.ylabel('Componente 2')
plt.grid()
plt.show()
